examples/slice/main.go

```diff
@@ -0,0 +1,18 @@
+package main
```

Comment:
Nice! Let's add a comment or two explaining what this program does, and ideally let's supply an example text file with it which contains suitable matching lines, and explain how to run the example on this data.